db/migrate/20241230153300_new_embeddings_tables.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class NewEmbeddingsTables < ActiveRecord::Migration[7.2]
+  def up
+    # We need "model_id" to be a bigint now that it's no longer a hardcoded ID.
+
+    create_table :ai_topics_embeddings, id: false do |t|
+      t.bigint :topic_id, null: false
+      t.bigint :model_id, null: false
+      t.integer :model_version, null: false
+      t.integer :strategy_id, null: false
+      t.integer :strategy_version, null: false
+      t.text :digest, null: false
+      t.column :embeddings, "halfvec", null: false
+      t.timestamps
+
+      t.index %i[model_id strategy_id topic_id],
+              unique: true,
+              name: "index_ai_topics_embeddings_on_model_strategy_topic"
+    end
+
+    create_table :ai_posts_embeddings, id: false do |t|
+      t.bigint :post_id, null: false
+      t.bigint :model_id, null: false
+      t.integer :model_version, null: false
+      t.integer :strategy_id, null: false
+      t.integer :strategy_version, null: false
+      t.text :digest, null: false
+      t.column :embeddings, "halfvec", null: false
+      t.timestamps
+
+      t.index %i[model_id strategy_id post_id],
+              unique: true,
+              name: "index_ai_posts_embeddings_on_model_strategy_post"
+    end
+
+    create_table :ai_document_fragments_embeddings, id: false do |t|
+      t.bigint :rag_document_fragment_id, null: false
+      t.bigint :model_id, null: false
+      t.integer :model_version, null: false
+      t.integer :strategy_id, null: false
+      t.integer :strategy_version, null: false
+      t.text :digest, null: false
+      t.column :embeddings, "halfvec", null: false
+      t.timestamps
+
+      t.index %i[model_id strategy_id rag_document_fragment_id],
+              unique: true,
+              name: "index_ai_fragments_embeddings_on_model_strategy_fragment"
+    end
+
+    # Copy data from old tables to new tables
```

Comment:
Don't we need to redo this again in a post migrate migration for embeddings that were created mid deploy ?

Reply:
I believe the batch job will take care of those. Will confirm.